app/repositories/s3_enrolment_repo.py

```diff
@@ -0,0 +1,107 @@
+import json
+from typing import Any, List
+
+import boto3
+
+from app.config import settings
+from app.domain.entities.enrolment_request import (
+    VALID_STATE_CHANGES,
+    EnrolmentFilters,
+    EnrolmentRequest,
+)
+from app.repositories.enrolment_repo import EnrolmentRepo
+from app.utils.error_handling import handle_s3_errors
+
+
+class S3EnrolmentRepo(EnrolmentRepo):
+    s3: Any
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.s3 = boto3.client("s3", **settings.s3_configuration)
+
+    def get_enrolments(
+        self, enrolment_filters: EnrolmentFilters
+    ) -> List[EnrolmentRequest]:
+        enrolment_filters = enrolment_filters.dict()
+        with handle_s3_errors():
+            enrolment_objects_list = self.s3.list_objects(
+                Bucket=settings.ENROLMENT_BUCKET
+            )
+        enrolments_list = []
+        for enrolments_object in enrolment_objects_list.get("Contents", []):
+            with handle_s3_errors():
+                obj = self.s3.get_object(
+                    Key=f"enrolments/{enrolments_object['Key']}.json",
```

Comment:
folder name should be enrolment_requests, same for bucket

Reply:
I'll use the enrolments that have been sent from TP-Enrolments. So, I'll update it with that repo's bucket.